util/crc32c_ppc.c

```diff
@@ -7,6 +7,7 @@
 //  COPYING file in the root directory of this source tree.
```

Comment:
BSD is not compatible with patents. I'm wondering if it is okay for us to include this header in our code.

Reply:
@yiwu-arbug The BSD licenses were approved by facebook legal in the past, for earlier drops that @kamasubb has submitted.  

Links to prior discussion on that topic:

https://github.com/facebook/rocksdb/commit/e612e317409e8a9d74cf05db0bd733403305f459
https://github.com/facebook/rocksdb/pull/2353
https://github.com/facebook/rocksdb/pull/2716

The pull requests show the discussion.  The commit shows 2716 was merged with these BSD licenses previously.  This new drop provides a performance enhancement to that previous crc work if the buffer to be crc'ed is zero filled.  The caller indicates the buffer is zero filled by passing in a null data pointer and a non-zero buffer length.